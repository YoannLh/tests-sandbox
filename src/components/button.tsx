import { useState } from 'react'

export function Button() {
  const [buttonLabel, setButtonLabel] = useState('Se connecter')
  return (
    <button
      style={{ margin: 'auto', border: '1px solid white' }}
      onClick={() => {
        setButtonLabel('Chargement...')
      }}
    >
      {buttonLabel}
    </button>
  )
}
