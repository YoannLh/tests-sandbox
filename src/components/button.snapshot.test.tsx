import { render } from '@testing-library/react'

import { Button } from './button'

it('renders correctly', () => {
  const { asFragment } = render(<Button />)
  expect(asFragment()).toMatchSnapshot()
})
