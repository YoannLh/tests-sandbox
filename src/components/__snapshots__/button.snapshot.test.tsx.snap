// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders correctly 1`] = `
<DocumentFragment>
  <button
    style="margin: auto; border: 1px solid white;"
  >
    Se connecter
  </button>
</DocumentFragment>
`;
